File: apps/web/lib/format.ts
// Capitalizes the first letter of a unit name for display (e.g. "kilometer" -> "Kilometer").
export function capitalize(text: string): string {
  return text.charAt(0).toUpperCase() + text.slice(1);
}

// Rounds a conversion result to a reasonable precision and trims trailing zeros
// (e.g. 609.60000000000002 -> "609.6").
export function formatNumber(value: number): string {
  return Number(value.toFixed(4)).toString();
}

File: src/unit_converter/web/format.py
def capitalize(text: str) -> str:
    """Capitalizes the first letter of a unit name (e.g. "kilometer" -> "Kilometer")."""
    return text[:1].upper() + text[1:]


def format_number(value: float) -> str:
    """Rounds to 4 decimals and trims trailing zeros (e.g. 609.6000...2 -> "609.6")."""
    text = f"{value:.4f}".rstrip("0").rstrip(".")
    if text in ("", "-0"):
        return "0"
    return text
